.github/actions/deploy_logcollection/action.yml

```diff
@@ -0,0 +1,66 @@
+name: Log Collection Deployment
+description: Deploy log collection functionality to the cluster.
+
+inputs:
+  logstash-port:
+    description: "The port of the logstash service."
+    default: "5045"
+  kubeconfig:
+    description: "The kubeconfig of the cluster to deploy to."
+    required: true
+  opensearch-user:
+    description: "The username of the opensearch cluster."
+    required: true
+  opensearch-pwd:
+    description: "The password of the opensearch cluster."
+    required: true
+  test:
+    description: "The e2e test payload."
+    required: true
+  provider:
+    description: "The CSP of the cluster."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Template Logcollection Helm Values
+      id: template
+      shell: bash
+      run: |
+        bazel run //hack/logcollector template -- \
+         --dir $(realpath .) \
+         --username ${{ inputs.opensearch-user }} \
+         --password ${{ inputs.opensearch-pwd }} \
+         --port ${{ inputs.logstash-port }} \
+         --info is-debug-cluster=false \
```

Comment:
As you want to deploy it for debug cluster to debug log collection, this should come from input.